include/alpaka/vec/Vec.hpp

```diff
@@ -582,32 +526,16 @@ namespace alpaka
             return
                 createVecFromIndexedFn<
                     TDim,
-                    detail::CreateLessEqual>(
+                    detail::CreateVecFromTwoIndexedVecsLessEqual>(
                         p,
                         q);
         }
 
         namespace detail
         {
-            //#############################################################################
-            //! A function object that returns the element-wise greater than or equal relation of two vectors.
             template<
                 std::size_t Tidx>
-            struct CreateGreaterEqual
-            {
-                //-----------------------------------------------------------------------------
-                ALPAKA_NO_HOST_ACC_WARNING
-                template<
-                    typename TDim,
-                    typename TVal>
-                ALPAKA_FN_HOST_ACC static auto create(
-                    Vec<TDim, TVal> const & p,
-                    Vec<TDim, TVal> const & q)
-                -> bool
-                {
-                    return p[Tidx] >= q[Tidx];
-                }
-            };
+            using CreateVecFromTwoIndexedVecsGreaterEqual = CreateVecByApplyingBinaryFnToTwoIndexedVecs<std::greater_equal, Tidx>;
```

Comment:
I would say that `std::*` functors will not work here because they can not be used on all devices e.g. CUDA/HIP.

Reply:
They should be constexpr with c++14 which would allow us to use them in device code.

Reply:
But the operation (functor) evaluation is runtime not compile time.

Reply:
As long as the functor is constexpr (which it is), there should be no problem to evaluate it at runtime.

Reply:
I am still confused. `constexpr` only means that the function can be evaluated at compile time with compile time vaules (input). This is here not the case.
Do we have a test where we create somewhere on the device a vector where the generator with a functor from `std::*` is used?

Reply:
Mhh each `+` operation on the device should use this generator. I am still confused how the compiler can generate device code e.g. for CUDA out of an host functor.

Reply:
What I am relying on here is `--expt-relaxed-constexpr` which we have enabled by default.

Reply:
Normally the compiler perform the `constexpr` operation during the compile time but maybe it works with runtime values too.
I will check your changes with HIP-clang too, to avoid that we remove required code.

Reply:
HIP-clang work!